extension/vector/src/index/hnsw_index_utils.cpp

```
@@ -12,41 +12,62 @@
+    const transaction::Transaction* transaction, const catalog::TableCatalogEntry* tableEntry,
+    const std::string& indexName) {
+    return catalog::Catalog::Get(context)->containsIndex(transaction, tableEntry->getTableID(),
+        indexName);
+}
+
+bool HNSWIndexUtils::validateIndexExistence(const main::ClientContext& context,
     const catalog::TableCatalogEntry* tableEntry, const std::string& indexName,
-    IndexOperation indexOperation) {
+    IndexOperation indexOperation, common::ConflictAction conflictAction) {
     auto transaction = transaction::Transaction::Get(context);
     switch (indexOperation) {
     case IndexOperation::CREATE: {
-        if (catalog::Catalog::Get(context)->containsIndex(transaction, tableEntry->getTableID(),
-                indexName)) {
-            throw common::BinderException{common::stringFormat(
-                "Index {} already exists in table {}.", indexName, tableEntry->getName())};
+        if (indexExists(context, transaction, tableEntry, indexName)) {
+            switch (conflictAction) {
+            case common::ConflictAction::ON_CONFLICT_THROW:
+                throw common::BinderException{common::stringFormat(
+                    "Index {} already exists in table {}.", indexName, tableEntry->getName())};
+            case common::ConflictAction::ON_CONFLICT_DO_NOTHING:
+                return true;
+            case common::ConflictAction::INVALID:
+                KU_UNREACHABLE;
+            }
         }
+        return false;
     } break;
     case IndexOperation::DROP:
     case IndexOperation::QUERY: {
-        if (!catalog::Catalog::Get(context)->containsIndex(transaction, tableEntry->getTableID(),
-                indexName)) {
-            throw common::BinderException{common::stringFormat(
-                "Table {} doesn't have an index with name {}.", tableEntry->getName(), indexName)};
+        if (!indexExists(context, transaction, tableEntry, indexName)) {
```

If this is query vector index, we definitely need to throw an exception.

---

validateIndexExistence is called in QUERY with the default ConflictAction value of THROW (since there isn't a parameter to change it), but I'll separate the cases for clarity